task1.js

```diff
@@ -0,0 +1,116 @@
+export class EventEmitter {
+  listeners = {};  // key-value pair
+
+  addListener(eventName, fn) {
+    if (Array.isArray(this.listeners[eventName])) {
+      this.listeners[eventName].push({fn: fn, isOnce: false});
+    } else {
+      this.listeners[eventName] = [{fn: fn, isOnce: false}];
+    }
+  }
+
+  on(eventName, fn) {
+    if (Array.isArray(this.listeners[eventName])) {
+      this.listeners[eventName].push({fn: fn, isOnce: false});
+    } else {
+      this.listeners[eventName] = [{fn: fn, isOnce: false}];
+    }
+  }
+
+  removeListener(eventName, fn) {
+    for (const listenerIndex in this.listeners[eventName]) {
+      if (this.listeners[eventName][listenerIndex].fn === fn) {
+        delete this.listeners[eventName].splice(listenerIndex, 1);
+      }
+    }
+  }
+
+  off(eventName, fn) {
+    for (const listenerIndex in this.listeners[eventName]) {
+      if (this.listeners[eventName][listenerIndex].fn === fn) {
+        delete this.listeners[eventName].splice(listenerIndex, 1);
+      }
+    }
```

Comment:
This code dulicates implementation of `removeListener`. You may use one of the method inside of the other to avoid code duplication.